FindOnPage.md

```diff
@@ -0,0 +1,664 @@
+    // and stopping the find operation (StopFind) are assumed to be handled by
+    // custom UI elements and user interaction, not directly in code here.
+    // User could then connect functions such as FindNext, FindPrevious, and StopFind
+    // to corresponding custom UI elements.
+
+    return true;
+}
+//! [ExecuteFindWithCustomUI]
+```
+#### .NET C#
+```csharp
+//! [ConfigureAndExecuteFindWithDefaultUI]
+async Task ConfigureAndExecuteFindWithDefaultUIAsync(string findTerm)
+{
+    try
+    {
+        // Check if the webView is already initialized and is an instance of CoreWebView2.
+        if (webView.CoreWebView2 == null)
+        {
+            throw new InvalidOperationException("WebView2 is not initialized.");
+        }
+
+        // Initialize the find configuration with specified settings.
+        var findConfiguration = new CoreWebView2FindConfiguration
+        {
+            FindTerm = findTerm,
+            IsCaseSensitive = false,
+            ShouldMatchWord = false,
+            FindDirection = CoreWebView2FindDirection.Forward
+        };
+
+        // By default Find will use the default UI and highlight all matches. If you want different behavior
+        // you can change the SuppressDefaultDialog and ShouldHighlightAllMatches properties here.
+
+        // Start the find operation with the specified configuration.
+        await webView.CoreWebView2.Find.StartFindAsync(findConfiguration);
+
+        // End user interaction is handled via UI.
+    }
+    catch (Exception ex)
+    {
+        // Handle any errors that may occur during the find operation.
+        Console.WriteLine($"An error occurred: {ex.Message}");
+    }
+}
+//! [ConfigureAndExecuteFindWithDefaultUI]
+```
+
+```csharp
+//! [ConfigureAndExecuteFindWithCustomUI]
+async Task ConfigureAndExecuteFindWithCustomUIAsync(string findTerm)
+{
+    try
+    {
+        // Check if the webView is already initialized and is an instance of CoreWebView2.
+        if (webView.CoreWebView2 == null)
+        {
+            throw new InvalidOperationException("WebView2 is not initialized.");
+        }
+
+        // Initialize the find configuration with specified settings.
+        var findConfiguration = new CoreWebView2FindConfiguration
+        {
+            FindTerm = findTerm,
+            IsCaseSensitive = false,
+            ShouldMatchWord = false,
+            FindDirection = CoreWebView2FindDirection.Forward
+        };
+
+        // Specify that a custom UI will be used for the find operation.
+        webView.CoreWebView2.Find.SuppressDefaultDialog = true;
+        webView.CoreWebView2.Find.ShouldHighlightAllMatches = true;
+
+        // Start the find operation with the specified configuration.
+        await webView.CoreWebView2.Find.StartFindAsync(findConfiguration);
+        // It's expected that the custom UI for navigating between matches (next, previous)
+        // and stopping the find operation will be managed by the developer's custom code.
```

Comment:
Does the custom code have the ability to push the ActiveMatchIndex back into the WebView? (So that other plugins can read it / be notified when it changes?) Or is the idea that the custom UI should just respond to the active match but not try to change it? When the user hits "Next" in the custom UI, the custom UI should perform a StartFind?

I think an example of a custom UI would help.

Reply:
+1 to a sample. User hits "Next" should call `FindNext`. Moving to an explicit index isn't supported in Edge today either

Reply:
That feature (setting the ActiveMatchIndex) is not currently supported. We can look at supporting that in the future if necessary.